From c68049a0f7f5b7add7d9691f5a94dc8e885a9fc1 Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Fri, 06 Jun 2025 13:29:05 +0300
Subject: [PATCH] [SW-228856] fit: call with original name

Context:
We encountered several code duplications related to
fit_foreach_property.
So we created a new fit services infrastructure
to solve this issue in a clean way.
The main Lego pieces are the newly created files:
fit_services.c/h

Current commit:
In this current commit, we invoke the function
fit_foreach_property with its original name
and not with its temporary name _fit_foreach_property.

General remark:
This infrastructure is intended to support ANY of the
fit related functions that we find today in fit_utils.c
or fit.c
Therefore, from now, please, anyone doing a change related
to the fit files should carefully double-check:
Any unified code can be (and should be) located in
fit_services.c/h
For this, please, follow the example of how it has been
done with fit_foreach_property.

Change-Id: I75016d2a15cd1d8d9451f7aaf676c66830cb150a
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/lib/module/module_lkd.c b/lib/module/module_lkd.c
index ce24552..dbdecfa 100644
--- a/lib/module/module_lkd.c
+++ b/lib/module/module_lkd.c
@@ -309,7 +309,7 @@
 	int rc = 0;
 
 	fit_required_device_name = NULL;
-	_fit_foreach_property((void *)(desc->base), "/", match_fit_prop_device, HL_FIT_DEVICE_PROP);
+	fit_foreach_property((void *)(desc->base), "/", match_fit_prop_device, HL_FIT_DEVICE_PROP);
 	if (fit_required_device_name) {
 		if (memcmp(fit_required_device_name, HL_DEVICE_TYPE, strlen(HL_DEVICE_TYPE))) {
 			LOG_ERR("Attempt to load the FIT, that is intended for device %s",
